src/App.jsx

```diff
@@ -14,33 +17,90 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function createInternalArrayOfObjectsWithId(arr) {
+  const internalArrayOfObjects = [];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  for (let i = 0; i < arr.length; i += 1) {
+    internalArrayOfObjects.push({
+      name: arr[i],
+      internalId: i,
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return internalArrayOfObjects;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+const goodsListWithId = createInternalArrayOfObjectsWithId(goodsFromServer);
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ABC = 'abc';
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  let sortedGoods = goodsListWithId.toSorted((good1, good2) => {
+    switch (sortValue) {
+      case SORT_BY_LENGTH:
+        return good1.name.length - good2.name.length;
+      case SORT_BY_ABC:
+        return good1.name.localeCompare(good2.name);
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods = sortedGoods.toReversed();
```

Comment:
The method `toReversed` does not exist on JavaScript arrays. Use the `reverse` method instead. Remember that `reverse` mutates the original array, so create a copy of the array if immutability is important.